tomcat/src/main/java/org/apache/coyote/http11/request/RequestURL.java

```diff
@@ -0,0 +1,98 @@
+package org.apache.coyote.http11.request;
+
+import org.apache.coyote.http11.HttpMethod;
+import org.apache.coyote.http11.exception.InvalidRequestLineException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RequestURL {
+    private static final int METHOD_INDEX = 0;
+    private static final int URL_INDEX = 1;
+    private static final int VERSION_INDEX = 2;
+    private static final String QUERY_STRING_DELIMITER = "?";
+    private static final String QUERY_PARAM_DELIMITER = "&";
+    private static final String KEY_VALUE_DELIMITER = "=";
+    private static final String NON_RESERVED_QUERY_PARAM_DELIMITER = "\\?";
+    private static final String NON_RESERVED_EXTENSION_DELIMITER = "\\.";
+    private static final String DEFAULT_EXTENSION = ".html";
+    private static final String EXTENSION_DELIMITER = ".";
+    private static final String SPACE_DELIMITER = " ";
+
+    private final HttpMethod method;
+    private final String url;
+    private final String version;
+    private final Map<String, String> requestParam;
+
+    private RequestURL(final HttpMethod method, final String url, final String version, final Map<String, String> requestParam) {
+        this.method = method;
+        this.url = url;
+        this.version = version;
+        this.requestParam = requestParam;
+    }
+
+    public static RequestURL from(String line) {
+        validate(line);
+
+        final String[] splitLine = line.split(SPACE_DELIMITER);
+
+        return new RequestURL(getMethod(splitLine[METHOD_INDEX]), splitLine[URL_INDEX],
+                splitLine[VERSION_INDEX], parseQueryParam(splitLine[URL_INDEX]));
+    }
+
+    private static void validate(final String line) {
+        if (line.isBlank() && line.isEmpty()) {
+            throw new InvalidRequestLineException();
+        }
+    }
+
+    private static Map<String, String> parseQueryParam(String url) {
+        if (!url.contains(QUERY_STRING_DELIMITER)) {
+            return Collections.emptyMap();
+        }
+        final String[] splitedQueryString = url.split(NON_RESERVED_QUERY_PARAM_DELIMITER)[1].split(QUERY_PARAM_DELIMITER);
+        Map<String, String> queryStrings = new HashMap<>();
+        for (String queryString : splitedQueryString) {
+            final String[] splitQuery = queryString.split(KEY_VALUE_DELIMITER);
+            queryStrings.put(splitQuery[0], splitQuery[1]);
+        }
+
+        return queryStrings;
+    }
+
+    public String getAbsolutePath() {
+        String absolutePath = url;
+        if (absolutePath.contains(QUERY_STRING_DELIMITER)) {
+            absolutePath = absolutePath.split(NON_RESERVED_QUERY_PARAM_DELIMITER)[0];
+        }
+        if (!absolutePath.contains(EXTENSION_DELIMITER)) {
+            absolutePath += DEFAULT_EXTENSION;
```

Comment:
```suggestion
absolutePath += EXTENSION_DELIMITER + HttpContentType.HTML.getContentType()
```

HttpContentType 에 있는 HTML 을 재사용해보는 건 어떨까요오?? 😁